WalletWasabi/Transactions/AllTransactionStore.cs

```diff
@@ -0,0 +1,236 @@
+				if (Directory.Exists(networkIndependentTransactionsFolderPath))
+				{
+					var oldTransactionsFolderPath = Path.Combine(networkIndependentTransactionsFolderPath, Network.Name);
+					if (Directory.Exists(oldTransactionsFolderPath))
+					{
+						lock (Lock)
+						{
+							foreach (var filePath in Directory.EnumerateFiles(oldTransactionsFolderPath))
+							{
+								try
+								{
+									string jsonString = File.ReadAllText(filePath, Encoding.UTF8);
+									var allWalletTransactions = JsonConvert.DeserializeObject<IEnumerable<SmartTransaction>>(jsonString)?.OrderByBlockchain() ?? Enumerable.Empty<SmartTransaction>();
+									foreach (var tx in allWalletTransactions)
+									{
+										TryAddNoLock(tx);
+									}
+
+									// ToDo: Uncomment when PR is finished.
+									// File.Delete(filePath);
+								}
+								catch (Exception ex)
+								{
+									Logger.LogTrace(ex);
+								}
+							}
+
+							// ToDo: Uncomment when PR is finished.
+							// Directory.Delete(oldTransactionsFolderPath, recursive: true);
+						}
+					}
+
+					// ToDo: Uncomment when PR is finished.
+					//// If all networks successfully migrated, too, then delete the transactions folder, too.
+					//if (!Directory.EnumerateFileSystemEntries(networkIndependentTransactionsFolderPath).Any())
+					//{
+					//	Directory.Delete(networkIndependentTransactionsFolderPath, recursive: true);
+					//}
+				}
+			}
+			catch (Exception ex)
+			{
+				Logger.LogWarning("Backwards compatibility could not be ensured.");
+				Logger.LogWarning(ex);
+			}
+		}
+
+		private void TryAddNoLock(SmartTransaction tx)
+		{
+			var hash = tx.GetHash();
+
+			if (tx.Confirmed)
+			{
+				if (MempoolStore.TryRemove(hash, out SmartTransaction found))
+				{
+					found.Update(tx);
+					ConfirmedStore.TryAdd(found);
+				}
+				else
+				{
+					ConfirmedStore.TryAdd(tx);
+				}
+			}
+			else
+			{
+				if (ConfirmedStore.TryGetTransaction(hash, out SmartTransaction found))
+				{
+					found.Update(tx);
+				}
+				else
+				{
+					MempoolStore.TryAdd(tx);
+				}
+			}
+		}
+
+		public void TryAdd(SmartTransaction tx)
+		{
+			lock (Lock)
+			{
+				TryAddNoLock(tx);
+			}
+		}
+
+		public bool TryUpdate(SmartTransaction tx, out SmartTransaction originalTx)
+		{
+			var hash = tx.GetHash();
+			lock (Lock)
+			{
+				// Do Contains first, because it's fast.
+				if (ConfirmedStore.Contains(hash) && ConfirmedStore.TryGetTransaction(hash, out originalTx))
+				{
+					originalTx.Update(tx);
+					return true;
+				}
+				else if (MempoolStore.Contains(hash) && MempoolStore.TryGetTransaction(hash, out originalTx))
+				{
+					originalTx.Update(tx);
+
+					// If confirmed update the height.
+					if (originalTx.Confirmed && MempoolStore.TryRemove(hash, out originalTx))
+					{
+						ConfirmedStore.TryAdd(originalTx);
```

Comment:
This is a question. Shouldn't this be 
```c#
return ConfirmedStore.TryAdd(originalTx);
```

Reply:
We updated it before, so if `TryAdd` for some strange reason would give us `false`, then we'd lie to the consumer of the API that we didn't update it.

Reply:
But this points out to a bug. Update doesn't update the serialization, only the memory, which is no good.